src/ray/core_worker/core_worker.h

```diff
@@ -1886,6 +1886,10 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
 
   std::atomic<bool> is_shutdown_ = false;
 
+  /// Whether the `Exit` function has been called, to avoid executing the exit
+  /// process multiple times.
+  std::atomic<bool> is_exit_ = false;
```

Comment:
drop a comment about the relationship between this and `is_shutdown_`

in general, in cases where we have confusing duplicate behaviors we should document it as clearly as possible to help future readers

Reply:
Added